HabitTracker.kmakai/HabitTracker.kmakai/HabitTracker.cs

```diff
@@ -0,0 +1,350 @@
+        {
+            Console.WriteLine("Please enter a valid input!");
+            index = int.Parse(TrackerMenu.ManageHabitsMenu(HabitsList)) - 1;
+        }
+
+        Habit habit = HabitsList[index];
+        while (true)
+        {
+            string option = TrackerMenu.HabitMenu(habit);
+            switch (option)
+            {
+                case "1":
+                    AddEntry(habit);
+                    break;
+                case "2":
+                    EditEntry(habit);
+                    break;
+                case "3":
+                    DeleteEntry(habit);
+                    break;
+                case "4":
+                    ViewAllEntries(habit);
+                    break;
+                case "0":
+                    TrackerMenu.ManageHabitsMenu(HabitsList);
+                    return;
+                default:
+                    Console.WriteLine("Please enter a valid input!");
+                    option = TrackerMenu.HabitMenu(habit);
+                    break;
+            }
+        }
+    }
+
+    private void ViewAllEntries(Habit habit)
+    {
+        Console.Clear();
+        Console.WriteLine($"All {habit.Name} entries:");
+        using (var connection = new SQLiteConnection(ConnectionString))
+        {
+            connection.Open();
+            var tableCommand = connection.CreateCommand();
+
+            tableCommand.CommandText = @$"SELECT * FROM {habit.Name}";
+
+            SQLiteDataReader reader = tableCommand.ExecuteReader();
+
+            while (reader.Read())
+            {
+                Console.WriteLine($"Id: {reader.GetInt32(0)} Date: {reader.GetString(1)} {habit.UnitOfMeasurement}: {reader.GetDouble(2)}");
+            }
+
+            connection.Close();
+        }
+        Console.WriteLine("-------------------------------------------\n");
+        Console.WriteLine("Press any key to continue...");
+        Console.ReadKey();
+        return;
+
+    }
+
+    private void DeleteEntry(Habit habit)
+    {
+        ViewAllEntries(habit);
+        int id = int.Parse(TrackerMenu.GetOption("Please enter the Id of the entry to delete"));
+
+        Console.WriteLine("Are you sure you want to delete this entry? (y/n)");
+        string option = Console.ReadLine().ToLower();
+        while (option != "y" && option != "n")
+        {
+            Console.WriteLine("Please enter a valid input!");
+            option = Console.ReadLine().ToLower();
+        }
+
+        if (option == "n") TrackerMenu.HabitMenu(habit);
+
+        using (var connection = new SQLiteConnection(ConnectionString))
+        {
+            connection.Open();
+            var tableCommand = connection.CreateCommand();
+
+            tableCommand.CommandText = @$"DELETE FROM {habit.Name} WHERE Id = {id}";
+
+            tableCommand.ExecuteNonQuery();
+
+            connection.Close();
+        }
+
+        Console.WriteLine($"Entry {id} deleted successfully!");
+
+        TrackerMenu.HabitMenu(habit);
```

Comment:
There is no need to call this method again as it will be called in the `ManageHabits()` method once you are done with this method. Same with the `EditEntry()` and the `AddEntry()` methods